src/unistd.rs

```diff
@@ -1132,6 +1132,58 @@ pub fn isatty(fd: RawFd) -> Result<bool> {
     }
 }
 
+
+/// Flags for `linkat` function.
+#[derive(Clone, Copy, Debug)]
+pub enum LinkatFlags {
+    SymlinkFollow,
+    NoSymlinkFollow,
+}
+
+/// Link one file to another file
+///
+/// Creates a new link (directory entry) at `newpath` for the existing file at `oldpath`. In the 
+/// case of a relative `oldpath`, the path is interpreted relative to the directory associated 
+/// with file descriptor `olddirfd` instead of the current working directory and similiarly for 
+/// `newpath` and file descriptor `newdirfd`. In case `flag` is LinkatFlags::SymlinkFollow and 
+/// `oldpath` names a symoblic link, a new link for the target of the symbolic link is created.
+/// The function shall fail if either olddirfd or newdirfd is None, returning ENOENT and EEXIST 
```

Comment:
WTF?  No it shall not fail.  That's not what will happen.

Reply:
> WTF? No it shall not fail. That's not what will happen.

What should happen? The spec I am referring to is here: http://pubs.opengroup.org/onlinepubs/9699919799/functions/linkat.html 

Failing is language the spec uses in the ERRORS section to describe errors returned. I.E.

> ERRORS
> These functions shall fail if:
> [EEXIST]
> [ENOENT]

When I tested by inserting "None" at the calls ENOENT and EEXIST were returned. 

Please comment on what you think should happen. If failing with ENOENT and EEXIST is not correct behavior then an update to the patch is needed.

Johnnie

Reply:
What _should_ happen when `olddirfd` is `None` is that you call `libc::linkat` with the `AT_FDCWD` argument, which the operating system interprets to mean that `oldpath` is relative to the current directory.  Similarly with `newdirfd`.

Reply:
@asomers Ok .. I see, thanks. I saw this mentioned and read the spec to mean this special value could be passed in directly by the user. However in case of None, instead of an returning an error, we could define behavior such that this value is used in this case as well. I'll make this update and use newdirfd as a reference. Thanks.

Reply:
No, Johnnie.  The code that you wrote *already uses `AT_FDCWD`*.  How did you not notice that?

Reply:
@asomers Thanks for the suggestion. The comment has been updated. Perhaps the patch is good now, but let me know if you'd like more changes. Thanks.

Reply:
@asomers Thanks for the suggestion. The comment has been updated. Perhaps the patch is good now, but let me know if you'd like more changes. Thanks.

Reply:
Hi @asmoers .. I am not sure the etiquette concerning who marks "Resolve conversation" for change requests such as this. Do you have more comment for this request?